compiler/ilgen/TypeDictionary.cpp

```diff
@@ -285,6 +286,121 @@ StructType::getFieldSymRef(const char *fieldName)
    return (TR::IlReference *)symRef;
    }
 
+class UnionType : public TR::IlType
+   {
+public:
+   TR_ALLOC(TR_Memory::IlGenerator)
+
+   UnionType(const char *name, TR_Memory* trMemory) :
+      TR::IlType(name),
+      _firstField(0),
+      _lastField(0),
+      _size(0),
+      _closed(false),
+      _symRefBV(4, trMemory),
+      _trMemory(trMemory)
+      { }
+
+   TR::DataType getPrimitiveType()                 { return TR::Address; }
+   void Close();
+
+   void AddField(const char *name, TR::IlType *fieldType);
+   TR::IlType * getFieldType(const char *fieldName);
+
+   TR::SymbolReference *getFieldSymRef(const char *name);
+   bool isUnion() { return true; }
```

Comment:
Technically, it doesn't need to be marked as virtual because `OMR::IlType::isUnion()` already is virtual :stuck_out_tongue_closed_eyes: 

For clarity though (and to keep things consistent), I'll mark it as virtual explicitly.